### PR TITLE
Replace semver with version compare module that supports composer ranges

### LINKED ISSUE #35: For Shopware 6 plugins: Version check does not support composer version ranges
When uploading a plugin that specifies it's compatibility to `shopware/core` with a version range like: "~6.3.0.2" the scs-commander fails with an error message.

To fix this, the npm package `semver` can't be used because the syntax for composer is slightly different and also `semver` does not support 4 version digits.

We need to implement our own version constraint checking.

### PR DESCRIPTION
**Pull request by magraina**
### Summary

Replace semver with new version compare module which supports composer version contraints

---

### PR checklist

- [x] I added changelogs to all plugins that are affected by this PR. _Or I did not add them on purpose._
- [x] I have written tests that fail without my changes. _Or I did not add them on purpose._
- [x] My code changes are pre-approved via live demonstration if it was possible.
  - [ ] Implications for the _Shopify Integration_ have been considered.
- [x] I have included instructions on how to test this, assuming they are not already apparent from the linked issue.

---

### Closes/Fixes/Related to

* Closes [#35](https://github.com/pickware/scs-commander/issues/35)

### CODE REVIEW

**Comment by magraina** [comment]
> [!NOTE]  
CI will fail as long as the module is not published

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## NEXT PATCH RELEASE
+
+### Bug fixes
+
+* Fixes version check does not support composer version ranges.
```
Comment (svenmuennich) [inline CHANGELOG.md]:
```suggestion
* Fixes plugin binary version checks to support composer version constraints.
```

File: lib/commands/compatibility.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 const Chalk = require('chalk');
 const Program = require('commander');
-const semver = require('semver');
+const { compareVersions, compareVersionsWithComperator } = require('composer-version-constraint-evaluator');
```
Comment (svenmuennich) [inline lib/commands/compatibility.js]: AFAIK the correct English word is "comparator".
```suggestion
const { compareVersions, compareVersionsWithComparator } = require('composer-version-constraint-evaluator');
```

File: package.json [hunk 3]
```diff
@@ -59,4 +59,4 @@
         "node": ">=18.0.0"
     },
     "preferGlobal": true
-}
+}
```
Comment (svenmuennich) [inline package.json]: What happened here? 😄

File: package.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
         "marked": "^2.0.3",
         "mz": "^2.4.0",
         "node-notifier": "^9.0.1",
-        "semver": "^7.3.5"
+        "composer-version-constraint-evaluator": "^1.0.0"
```
Comment (svenmuennich) [inline package.json]: We should scope the package name to pickware.
```suggestion
        "@pickware/composer-version-constraint-evaluator": "^1.0.0"
```

**Review by svenmuennich — CHANGES_REQUESTED** [review]

File: lib/commands/compatibility.js (hunk 2 above)
Comment (magraina) [reply lib/commands/compatibility.js]: Oh damn. Yes, you're right

File: package.json (hunk 4 above)
Comment (magraina) [reply package.json]: Haven't published yet. I did not know that we have such a scope

File: package.json (hunk 3 above)
Comment (magraina) [reply package.json]: I wonder how that has happend

**Review by svenmuennich — APPROVED** [review]

**Pull request closed without merging**